### PR TITLE
Add support for editing from context menu.

### PR DESCRIPTION
**Pull request by carloskelly13**
- Adds edit functionality from context menu
- Prevents malformed story data from coming to dashboard

### CODE REVIEW

File: assets/src/dashboard/components/cardGridItem/cardItemMenu.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ export default function CardItemMenu({
         className="grid-story-popover-menu"
         isOpen={contextMenuId === story.id}
         framelessButton
-        onSelect={onMenuItemSelected}
+        onSelect={(menuItem) => onMenuItemSelected(menuItem, story)}
```
Comment (BrittanyIRL) [inline assets/src/dashboard/components/cardGridItem/cardItemMenu.js:87]: nit: could you update storybook to see story as well just so we keep things current there?

**Review by BrittanyIRL — APPROVED** [review]
works great. one nit to keep storybook current.

File: assets/src/dashboard/components/cardGridItem/cardItemMenu.js (hunk 1 above)
Comment (carloskelly13) [reply assets/src/dashboard/components/cardGridItem/cardItemMenu.js:87]: Great call!

File: assets/src/dashboard/app/views/shared/storyGridView.js [hunk 2]
```diff
@@ -54,6 +55,14 @@ const StoryGridView = ({
   bottomActionLabel,
 }) => {
   const [contextMenuId, setContextMenuId] = useState(-1);
+
+  const handleMenuItemSelected = useCallback((sender, story) => {
+    setContextMenuId(-1);
+    if (sender.value === STORY_CONTEXT_MENU_ITEMS[0].value) {
```
Comment (mariano-formidable) [inline assets/src/dashboard/app/views/shared/storyGridView.js]: Nit, this isn't very clear what context menu item is being selected 😱.  What if we did the following:
```
// Create a CONTEXT_MENU_ACTIONS enum in constants.js like this:
export const CONTEXT_MENU_ACTIONS = {
  OPEN_IN_EDITOR: 'open-in-editor-action',
  ...,
}

// Update STORY_CONTEXT_MENU_ITEMS constant as follows:
export const STORY_CONTEXT_MENU_ITEMS = [
  {
    label: __('Open in editor', 'web-stories'),
    value: CONTEXT_MENU_ACTIONS.OPEN_IN_EDITOR,
  },
  ...,
]
```
And then we can use `CONTEXT_MENU_ACTIONS` in the if statement like so:
```
if (sender.value === CONTEXT_MENU_ACTIONS.OPEN_IN_EDITOR) {
  ...
}
```
And then it'll be pretty clear which action is being called (in fact, when we hook up the rest, we can just use a `switch` with each `case` being a `CONTEXT_MENU_ACTIONS` item).

**Review by mariano-formidable — CHANGES_REQUESTED** [review]

File: assets/src/dashboard/app/views/shared/storyGridView.js (hunk 2 above)
Comment (carloskelly13) [reply assets/src/dashboard/app/views/shared/storyGridView.js]: Good idea.

File: assets/src/dashboard/constants.js [hunk 3]
```diff
@@ -82,6 +82,15 @@ export const VIEW_STYLE = {
...
+export const STORY_CONTEXT_MENU_ACTIONS = {
+  OPEN_IN_EDITOR: 'open-in-editor-action',
+  PREVIEW: 'preview-action',
+  RENAME: 'rename-action',
+  DUPLICATE: 'duplicate-action',
+  CREATE_TEMPLATE: 'create-template-action',
+  DELETE: 'delete-story-action',
+};
+
 export const STORY_CONTEXT_MENU_ITEMS = [
```
Comment (mariano-formidable) [inline assets/src/dashboard/constants.js:94]: Nit, `STORY_CONTEXT_MENU_ITEMS` should be updated to use `STORY_CONTEXT_MENU_ACTIONS` just so we don't have two places defining the same string that could easily introduce a oneoff typo.  But not blocking!  Great job!!

**Review by mariano-formidable — APPROVED** [review]
One, non-blocking nit!  Great job!

File: assets/src/dashboard/constants.js (hunk 3 above)
Comment (carloskelly13) [reply assets/src/dashboard/constants.js:94]: I just noticed this. I will get this in the next one!

**Pull request merged**